service/lib/agama/dbus/storage/interfaces/filesystem.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2024] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "dbus"
+
+module Agama
+  module DBus
+    module Storage
+      module Interfaces
+        # Interface for file systems.
+        #
+        # @note This interface is intended to be included by {Device} if needed.
+        module Filesystem
+          FILESYSTEM_INTERFACE = "org.opensuse.Agama.Storage1.Filesystem"
+          private_constant :FILESYSTEM_INTERFACE
+
+          # File system type.
+          #
+          # @return [String] e.g., "ext4"
+          def filesystem_type
+            storage_device.filesystem.type.to_s
+          end
+
+          # Whether the file system contains an EFI.
```

Comment:
to be honest I find this explanation confusing  and prefer original one at https://github.com/yast/yast-storage-ng/blob/7157f8f78921ca688a41618f362ea2e04a0d4423/src/lib/y2storage/filesystems/base.rb#L225

EFI is Extensible Firmware Interface and ESP is EFI system partition, so later is what can file system contain.